### DISCUSSION TITLE
Customize CLI command names

### DISCUSSION THREAD

**Discussion by sagikazarmark** [dagger/dagger#12364] [Ideas]
### What are you trying to do?

Allow customizing how function names are translated into commands.

For example:

```
// +command=e2e
func (m *Ci) E2e() {
    // ...
}
```

### Why is this important to you?

Function names are often translated to weird commands. For example `E2e` becomes `e-2-e`.

### How are you currently working around this?

I call my end-to-end test function `Etoe`. 🙈

**Comment by helderco** [comment]
Rather than customizing the CLI name, we need to enable customizing the name in the API. Python and Typescript do that through the decorators:
- [Python: Name Overrides](https://docs.dagger.io/manuals/developer/python/558732/name-overrides)
- [TypeScript: Aliases](https://docs.dagger.io/manuals/developer/typescript/406314/aliases)

This would add paritity in Go, but probably with `+name` instead of `+command`.

However, if doing the same as Py or TS, does it lead to the same issue of `e2e` translating to `e-2-e`? I'll need to check where that's happening because we use two different Go utils in different places. One of those (I suspect _strcase_) is breaking numbers instead of just changes in casing.

The API uses the same naming convention as TypeScript (_camelCase_), which is similar to Go (_PascalCase_), in that both use casing to split words. Python's much simpler to break words into (_snake_case_), same as the CLI (_kebab-case_). So maybe using the CLI's convention is the more neutral and consistent way to override the names in all SDKs.

Think about acronyms. For example `SSH` translates to `s-s-h` unless you have a list of common acronyms (like we do). But overriding to `+name=ssh` would make it less ambiguous or easier to know how to convert consistently.
